arangod/Cluster/ClusterInfo.cpp

```diff
@@ -3813,7 +3836,11 @@ Result ClusterInfo::createCollectionsCoordinator(
       }
     }
 
-    if (nrDone->load(std::memory_order_acquire) == infos.size() &&
+    auto nrDone =
+        report->doUnderLock([](CreateCollectionReport const& report) -> size_t {
+          return report.nrDone;
+        });
+    if (nrDone == infos.size() &&
```

Comment:
If nrDone equal infos.size() it means info will not change.
So it's safe to touch it without lock after this condition

Reply:
In such case this code is incorrect, so if we will see data race here, we will found bug and will try to understand why it's repeated

Reply:
I do not understand the comment.
info.size() is constant and will not change (we do not suddenly create more collections then asked for)
But nrDone will change, everytime one of the collections shows up in current.
=> I need synchronization on nrDone, which achieved over the Guarded structure.
info.size() is indeed accessed without a lock.

Reply:
It's about 3918